src/drivers/ins/microstrain/MicroStrain.cpp

```diff
@@ -309,27 +474,35 @@ mip_cmd_result MicroStrain::configureImuMessageFormat()
 	uint16_t mag_decimation = base_rate / (uint16_t)_param_ms_mag_rate_hz.get();
```

Comment:
This is a bit of a weird situation, but converting hertz to decimation doesn't always work. First of all, if any hertz are set to 0, this would cause a divide by zero error, but also if you tried to stream 1000 hertz at 3 hertz, 1000/3 is not a whole number, so you would have to do some rounding, and the result would be that the driver would not work quite as expected. Take a look at [this function](https://github.com/LORD-MicroStrain/microstrain_inertial_driver_common/blob/e73418761d47b7a3bba6971e75cf9b895d4a4bac/src/utils/mip/ros_mip_device_main.cpp#L365) in the ROS driver